src/interpolation_functions.jl

```diff
@@ -0,0 +1,304 @@
+      old_x_n = truncate(x_n; cutoff)
+      x_n =
+        coeffs[i, j] * const_itn(s) +
+        (-1) * x_n1 +
+        poly_itn(s, [-2, 4]; dim=dims[1]) * old_x_n
+      x_n1 = old_x_n
+    end
+    coeff_j = truncate(x_n + poly_itn(s, [1, -2]; dim=dims[1]) * x_n1; cutoff)
+
+    # y_n-1 = c_n-1 - y_n+1 + (4y-2) * y_n
+    old_y_n = truncate(y_n; cutoff)
+    y_n = coeff_j + (-1) * y_n1 + poly_itn(s, [-2, 4]; dim=dims[2]) * old_y_n
+    y_n1 = old_y_n
+  end
+
+  return truncate(y_n + poly_itn(s, [1, -2]; dim=dims[2]) * y_n1; cutoff)
+end
+
+""" Helper function for function_itensornetwork (1D case) """
+function greatest_n(coeffs, n)
+  if n < 1
+    throw("n must be greater than 0")
+  end
+  sorted_indices = sortperm(coeffs; by=abs, rev=true)
+  top_n_indices = sorted_indices[1:min(n, length(coeffs))]
+  output = zeros(eltype(coeffs), length(coeffs))
+  output[top_n_indices] = coeffs[top_n_indices]
+  last_non_zero = findlast(output .!= 0)
+  if isempty(last_non_zero)
+    return []
+  end
+  return output[1:last_non_zero]
+end
+
+""" Helper function for function_itensornetwork (2D case) """
+function greatest_n_2d(coeffs_matrix, n)
+  if n < 1
+    throw("n must be greater than 0")
+  end
+  dims = size(coeffs_matrix)
+  coeffs = vec(coeffs_matrix)
+  sorted_indices = sortperm(coeffs; by=abs, rev=true)
+  top_n_indices = sorted_indices[1:min(n, length(coeffs))]
+  output = zeros(eltype(coeffs), length(coeffs))
+  output[top_n_indices] = coeffs[top_n_indices]
+  matrix = reshape(output, dims)
+  last_row, last_col = size(matrix)
+  while last_row > 0 && all(matrix[last_row, :] .== 0)
+    last_row -= 1
+  end
+  while last_col > 0 && all(matrix[:, last_col] .== 0)
+    last_col -= 1
+  end
+  return matrix[1:last_row, 1:last_col]
+end
+
+""" 
+    function_itensornetwork(s,f, cutoff=1e-3, max_coeffs=100, mode="fourier", by_mag=true)
+    Takes in a function `f` as a black box, outputs a tensor network approximation using Fourier or Chebyshev series. Supports 1D or 2D functions.
+
+    #Arguments
+    - `s`: IndsNetworkMap
+    - `f`: a real-valued function on domain [0,1] or [0,1]⊗[0,1] which can be given as a black box
+    - `cutoff`: (default=1e-3) specifies at what value should the fourier/chebyshev coefficients be truncated below
+    - `max_coeffs`: the maximum number of coefficients allowed in 1D. In 2D, it is the maximum degree of a coefficient in each dimension
+    - `mode`: ["fourier", "chebyshev"] "fourier" uses a fourier basis and "chebyshev" uses a chebyshev polynomial basis
+    - `by_mag`: boolean. if false, will use the `max_coeffs` coefficients of lowest degree. if true, will use the `max_coeffs` coefficients of greatest absolute magnitude.
+    - `trunc_level`: what cutoff the periodic truncation is done at
+"""
+function function_itensornetwork(
+  s::IndsNetworkMap,
+  f::Function;
+  cutoff::Float64=1e-3,
+  max_coeffs::Integer=100,
+  mode="fourier",
+  by_mag=false,
+  trunc_level=1e-16,
```

Comment:
The cutoff here is different than the standard `truncate` cutoff, I would rename (i.e. make `trunc_level` cutoff)

Reply:
renamed `trunc_level` -> `cutoff` and `cutoff` -> `chop_level` to disambiguate